src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -182,6 +182,30 @@ namespace winrt::TerminalApp::implementation
         _showDialogHandlers(*this, dialog);
     }
 
+    // Method Description:
+    // - Displays a dialog for warnings found while closing the terminal app using
+    // - key binding with multiple tabs opened. Display messages to warn user
+    // - that more than 1 tabs are opend, and once the user click the OK button, remove
+    // - all the tabs and shut down and app. If cancel is clicked, the dialog will close
+    // - Only one dialog can be visible at a time. If another dialog is visible
+    //   when this is called, nothing happens. See _ShowDialog for details
+    void TerminalPage::_ShowCloseWarningDialog()
+    {
+        // To do: change these strings to localized strings in resource loader
+        auto title = _resourceLoader->GetLocalizedString(L"CloseWindowWarningTitle");
+        auto primaryButtonText = _resourceLoader->GetLocalizedString(L"Close all");
+        auto closeButtonText = _resourceLoader->GetLocalizedString(L"Cancel");
+
+        Controls::ContentDialog dialog;
+        dialog.Title(winrt::box_value(title));
+
+        dialog.PrimaryButtonText(primaryButtonText);
+        dialog.CloseButtonText(closeButtonText);
```

Comment:
Should "Close All" be the highlighted (colored) button, or should the cancel button be the highlighted one? Should neither?

Reply:
Maybe for now we can follow the close warning of Edge - The two buttons are both "grey". One easy and quick fix is to use SecondaryButton instead of CloseButton.